activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/query/rest/TaskPermissionsHelper.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017-2020 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.activiti.cloud.services.query.rest;
+
+import org.activiti.api.runtime.shared.security.SecurityManager;
+import org.activiti.cloud.api.task.model.QueryCloudTask.TaskPermissions;
+import org.activiti.cloud.services.query.model.QTaskEntity;
+import org.activiti.cloud.services.query.model.TaskEntity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskPermissionsHelper {
+    
+    private final SecurityManager securityManager;
+
+    private final TaskControllerHelper taskControllerHelper;
+
+    public TaskPermissionsHelper(SecurityManager securityManager, TaskControllerHelper taskControllerHelper) {
+        this.securityManager = securityManager;
+        this.taskControllerHelper = taskControllerHelper;
+    }
+
+    public void setCurrentUserTaskPermissions(TaskEntity task) {
+        String userId = securityManager.getAuthenticatedUserId();
+        if(userId!=null) {
+            List<String> userGroups = securityManager.getAuthenticatedUserGroups();
+            List<TaskPermissions> permissions = new ArrayList<>();
+            if(!canUserViewTask(task)) return;
+            
+            permissions.add(TaskPermissions.VIEW);
+            
+            if(canUserClaimTask(task, userId, userGroups)) {
+                permissions.add(TaskPermissions.CLAIM);
+            }
+            
+            if(canUserReleaseTask(task, userId, userGroups)) {
+                permissions.add(TaskPermissions.RELEASE);
+            }
+            
+            if(isUserAssignee(task, userId)) {
+                permissions.add(TaskPermissions.UPDATE);
```

Comment:
Can you give owner update permissions too? By default current user is [set as owner when creating a new standalone task](https://github.com/Activiti/Activiti/blob/develop/activiti-core/activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskRuntimeImpl.java#L251) therefore owner should be able to update it later.